### PR TITLE
Use convention plugins instead of script plugins

### PR DESCRIPTION
**Pull request by jmatsu**
ref: #86 (but not close)

I'm planning to create a code-generation tool as a Gradle plugin, which should be robust for future updates of AGP and/or Gradle. I'm expecting the implementation of the code-generation tool would be so huge, so we need readability and maintainability of the tool's code. However, script plugins are not so readable for huge codebase and it's not unrealistic to create a new repository to maintain plugins for this repo. That's why we need convention plugins.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -46,8 +65,6 @@ jobs:
               --url 'https://deploygate.com/api/users/${{ secrets.SHARED_DEPLOYGATE_APP_OWNER_NAME_FOR_PUBLIC_REPO }}/apps' | \
               jq -r '"\(.results.package_name) \(.results.distribution.url)"'
           done < <(find ./sample/build/outputs/bundle -name "*.aab")
-      - name: Check javadocs
-        run: ./gradlew sdk:generateJavadocForReleasePublication sdkMock:generateJavadocForReleasePublication
```
Comment (jmatsu) [inline .github/workflows/test.yml:50]: Same verifications will be done in the `build-sdk` job above.

File: gradle.properties [hunk 2]
```diff
@@ -1 +1,2 @@
-android.useAndroidX=true
\ No newline at end of file
+android.useAndroidX=true
+android.disableAutomaticComponentCreation=true
```
Comment (jmatsu) [inline gradle.properties:2]: Default since AGP8

File: plugins/library/src/main/java/com/deploygate/plugins/SdkPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        buildFeatures {
+            buildConfig = true
+        }
+
+        lint {
+            abortOnError = false
+        }
+
+        publishing {
```
Comment (jmatsu) [inline plugins/library/src/main/java/com/deploygate/plugins/SdkPlugin.kt:124]: New DSL allows us to remove Javadoc related task declarations!

File: scripts/verify-publications.bash [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "${module_name}:verifyBytecodeVersionRelease" \
+    "${module_name}:publishReleasePublicationToMavenLocal" \
+    --stacktrace
+
+cat "${module_name}/build/publications/release/module.json"
+
+readonly artifact_id="$(jq -r '.component.module' "${module_name}/build/publications/release/module.json")"
+readonly version="$(jq -r '.component.version' "${module_name}/build/publications/release/module.json")"
+
+cat<<EOF > "${tmp_dir}/expected.txt"
```
Comment (jmatsu) [inline scripts/verify-publications.bash:24]: Check if all artifacts, which are javadoc, sources, and runtime, exist.

File: plugins/library/src/main/java/com/deploygate/plugins/SdkPlugin.kt [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+import com.deploygate.plugins.ext.libraryExtension
+import org.gradle.api.GradleException
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.dependencies
+
+class SdkPlugin : Plugin<Project> {
```
Comment (jmatsu) [inline plugins/library/src/main/java/com/deploygate/plugins/SdkPlugin.kt:14]: from `sdk.build.gradle`

File: plugins/library/src/main/java/com/deploygate/plugins/MavenPublishPlugin.kt [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+import org.gradle.api.tasks.Exec
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+import org.gradle.plugins.signing.SigningExtension
+import java.util.Locale
+
+class MavenPublishPlugin : Plugin<Project> {
```
Comment (jmatsu) [inline plugins/library/src/main/java/com/deploygate/plugins/MavenPublishPlugin.kt:22]: from `publishing.build.gradle`

**Review by satsukies — APPROVED** [review]
LGTM

**Comment by jmatsu** [comment]
Thanks~

**Pull request merged**